flow_prompt/ai_models/claude/claude_model.py

```diff
@@ -0,0 +1,180 @@
+from flow_prompt.ai_models.ai_model import AI_MODELS_PROVIDER, AIModel
+import logging
+
+from flow_prompt.responses import AIResponse
+from decimal import Decimal
+from enum import Enum
+
+import typing as t
+from dataclasses import dataclass
+
+from flow_prompt.ai_models.claude.responses import ClaudeAIReponse
+from openai.types.chat import ChatCompletionMessage as Message
+from flow_prompt.responses import Prompt
+from flow_prompt.settings import Secrets
+from flow_prompt.exceptions import RetryableCustomException
+import anthropic 
+
+logger = logging.getLogger(__name__)
+
+
+
+C_200K = 200000
+
+class FamilyModel(Enum):
+    haiku = "Claude 3 Haiku"
+    sonnet = "Claude 3 Sonnet"
+    opus = "Claude 3 Opus"
+
+DEFAULT_PRICING = {
+    "price_per_prompt_1k_tokens": Decimal(0.00025),
+    "price_per_sample_1k_tokens": Decimal(0.00125),
+}
+
+CLAUDE_AI_PRICING = {
+    FamilyModel.haiku.value: {
+        C_200K: {
+            "price_per_prompt_1k_tokens": Decimal(0.00025),
+            "price_per_sample_1k_tokens": Decimal(0.00125),
+        }
+    },
+    FamilyModel.sonnet.value: {
+        C_200K: {
+            "price_per_prompt_1k_tokens": Decimal(0.003),
+            "price_per_sample_1k_tokens": Decimal(0.015),
+        }
+    },
+    FamilyModel.opus.value: {
+        C_200K: {
+            "price_per_prompt_1k_tokens": Decimal(0.015),
+            "price_per_sample_1k_tokens": Decimal(0.075),
+        }
+    },
+}
+
+
+@dataclass(kw_only=True)
+class ClaudeAIModel(AIModel):
+    model: str
+    api_key: str = ""
+    provider: AI_MODELS_PROVIDER = AI_MODELS_PROVIDER.CLAUDE
+    family: str = None
+    
+    def __post_init__(self):
+        if 'haiku' in self.model:
+            self.family = FamilyModel.haiku.value
+        elif 'sonnet' in self.model:
+            self.family = FamilyModel.sonnet.value
+        elif 'opus' in self.model:
+            self.family = FamilyModel.opus.value
+        else:
+            logger.warning(
+                f"Unknown family for {self.model}. Please add it obviously. Setting as Claude 3 Haiku"
+            )
+            self.family = FamilyModel.haiku.value
+
+        logger.debug(f"Initialized ClaudeAIModel: {self}")
+        
+        secrets = Secrets()
+        self.api_key = secrets.CLAUDE_API_KEY
+        
+
+    def call(self, messages: t.List[dict], max_tokens: int, **kwargs) -> AIResponse:
+        
+        common_args = {
```

Comment:
it can be moved in  shared constants btw all models

Reply:
i'm sorry, what exactly can be moved to shared constants?

Reply:
I mean parts of common args, which are: top_p, temperature and all which were copypasted from openai model, as i see